src/Tribe/Repositories/Traits/Post_Tickets.php

```diff
@@ -0,0 +1,292 @@
+	 * As an example filtering by "USD" when Tribe Commerce tickets use the "EUR" code and
+	 * WooCommerce tickets use the "USD" code means "only fetch events that have WooCommerce
+	 * tickets".
+	 *
+	 * @since TBD
+	 *
+	 * @param string|array $symbol One or more currency symbols or currency ISO codes. E.g.
+	 *                             "$" and "USD".
+	 *
+	 * @throws Tribe__Repository__Void_Query_Exception If no provider uses the specified currency symbol
+	 *                                                 or ISO code.
+	 */
+	public function filter_by_cost_currency_symbol( $symbol ) {
+		$repo = $this;
+
+		// If the repo is decorated, use that.
+		if ( ! empty( $repo ) ) {
+			$repo = $this->decorated;
+		}
+
+		/** @var Tribe__Tickets__Commerce__Currency $currency */
+		$currency      = tribe( 'tickets.commerce.currency' );
+		$symbols       = (array) $symbol;
+		$request_codes = [];
+
+		/*
+		 * Transform the request symbols into ISO codes; due to its ambiguous nature a
+		 * symbol might match 0+ ISO codes.
+		 */
+		foreach ( $symbols as $request_symbol ) {
+			$request_codes[] = (array) $currency->get_symbol_codes( $request_symbol );
+		}
+		$request_codes = array_unique( call_user_func_array( 'array_merge', $request_codes ) );
+
+		if ( empty( $request_codes ) ) {
+			$reason = 'The specified currency symbol or ISO code is not supported.';
+			throw Tribe__Repository__Void_Query_Exception::because_the_query_would_yield_no_results( $reason );
+		}
+
+		// Compile a list of ticket providers that are active and use one of the requested ISO codes.
+		$request_providers = [];
+
+		if ( array_intersect( $request_codes, (array) $currency->get_currency_code() ) ) {
+			$request_providers[] = 'tpp';
+		}
+
+		$providers = [
+			'eddticket' => 'Tribe__Tickets_Plus__Commerce__EDD__Main',
+			'wooticket' => 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main',
+		];
+
+		foreach ( $providers as $slug => $provider ) {
+			if ( ! class_exists( $provider ) ) {
+				continue;
+			}
+
+			$provider_symbol = $currency->get_provider_symbol( $provider );
+
+			if ( ! is_string( $provider_symbol ) ) {
+				continue;
+			}
+
+			$provider_codes = (array) $currency->get_symbol_codes( $provider_symbol );
+
+			if ( array_intersect( $request_codes, $provider_codes ) ) {
+				// This provider uses one of the request ISO codes, use it.
+				$request_providers[] = $slug;
+			}
+		}
+
+		if ( empty( $request_providers ) ) {
+			$reason = 'No ticket provider uses the specified currency symbol or ISO code.';
+			throw Tribe__Repository__Void_Query_Exception::because_the_query_would_yield_no_results( $reason );
+		}
+
+		$providers_regex = sprintf( '(%s)', implode( '|', $request_providers ) );
+
+		global $wpdb;
+		$prefix = 'by_cost_currency_symbol_';
+		// Join to the meta that relates tickets to events but only for the providers that have the required symbols.
+		$repo->join_clause( "JOIN {$wpdb->postmeta} {$prefix}_ticket_event 
+			ON ( 
```

Comment:
:no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: 6a48c80ba8d68400883ce671703d4d1e</sub>